modules/RTC/RTC.js

```diff
@@ -287,18 +287,57 @@ RTC.prototype.createRemoteTrack = function (event) {
 
 /**
  * Removes all JitsiRemoteTracks associated with given MUC nickname (resource
- * part of the JID).
- * @param resource the resource part of the MUC JID
- * @returns {JitsiRemoteTrack|null}
+ * part of the JID). Returns array of removed tracks.
+ *
+ * @param {string} resource - The resource part of the MUC JID.
+ * @returns {JitsiRemoteTrack[]}
  */
 RTC.prototype.removeRemoteTracks = function (resource) {
-    var remoteTracks = this.remoteTracks[resource];
+    var remoteTracksForResource = this.remoteTracks[resource];
+    var removedTracks = [];
+
+    if (remoteTracksForResource) {
```

Comment:
Why do we need these changes? The old code seems simpler for me and I think both versions are doing the same thing.